api/src/main/java/org/apache/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -120,6 +124,9 @@ public R or(R leftResult, R rightResult) {
           return in(pred.ref(), pred.literal());
         case NOT_IN:
           return notIn(pred.ref(), pred.literal());
+        case STARTS_WITH:
+          /* startsWith accepts only Strings, hence type-casting */
+          return startsWith((BoundReference<String>) pred.ref(), (Literal<String>) pred.literal());
```

Comment:
I didn't catch this the first time through, but if the `startsWith` method above is parameterized by `T` then there shouldn't be a need to cast these to literals and bound references parameterized by `String`.

Reply:
`startsWith` was parameterised  with `T`, but I missed removing the casts here.